src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/NomisData.kt

```diff
@@ -372,6 +400,16 @@ interface NomisDataDsl {
     dsl: IncidentDsl.() -> Unit = {},
   ): Incident
 
+  @CSIPReportDslMarker
+  fun csipReport(
+    offender: Offender,
```

Comment:
I wonder if you could do this 
`offender: Offender = offenderBooking.offender`

it feels like you should be able to default at least of them. If you supply a booking you shouldn't really need to supply and offender

Maybe I don't undertstand why booking is optional in the JPA

Reply:
Equally, could a csipReport be a child of an offender (or for that matter a booking) rather then a top level entity?
In  fact wouldn't that make more sense or can you create a CSIP with no relationship with prisoner??

Reply:
I think you're right about child of offender.  Looks like the nullability of the offenderBooking means it can't be part of that (although no idea why both are on the table??)
Will make child of offender (obvious, now you've mentioned it)

Reply:
Will address issues in separate PR